src/guiguts/application.py

```diff
@@ -49,6 +49,28 @@ def __init__(self):
         maintext().focus_set()
         maintext().add_modified_callback(self.update_title)
 
+        self.set_auto_image(preferences["AutoImage"])
+
+    def get_auto_image(self):
```

Comment:
Python has a cool featured called properties that can streamline things like this to make them more intuitive to use, eg:
```python
@property
def auto_image(self):
    return preferences["AutoImage"]

@auto_image.settr
def auto_image(self, value):
    preferences["AutoImage"] = value
    statusbar().set("see img", "Auto Img" if value else "See Img")
    if value:
        self.auto_image_check()
```
Then you can use them like:
```python
self.auto_image = not self.auto_image
```

Reply:
Also I need to look at the preferences class a bit closer as this seems like a bit of an odd pass-through but clearly we are taking action on changes not just changing the value.

Reply:
Funny, cos I was using property for this and elsewhere, but I was concerned the side effects of this one were too big to be a good use of properties, so reverted to regular methods. Is it OK to use properties  when there are significant side effects? Thanks.

Reply:
IMHO this is a fine use-case for properties. I'd probably look askance at something that was doing something potentially high-latency (external system calls, network traffic, etc) because I suspect it would be a surprise to a developer to have an get/set take a "long time" but this feels reasonable -- particularly since the only non-setter stuff happening are UI changes and this is a UI app. In general it's a better practice in python to use properties over getters/setters.

**However** one could easily that a property should do nothing that isn't necessary to get/set the attribute and activities like calls to update the UI violate that, and this clearly falls into that case.

Reply:
Thanks. I'll revert to using property along those lines. It was when I added the call to the repeating image check, I bottled out. Good to know that "reasonable" size effects are ok, thanks.

Reply:
You bet. And apologies for mansplaining properties to you ☹️

Reply:
Not at all. You've added to my understanding, as usual. 
And of course I have no experience to rely on about where to draw the line or how to organise things pythonically, so please keep telling me stuff, even if you suspect I might have seen it in a tutorial.
I'm sure when you or other experienced devs get a chance to look at other stuff I've already done, you'll see quite a lot to improve. I keep thinking of things, even as I use what I've already done, so I'm sure you will.